pkg/daemon/ceph/client/filesystem.go

```diff
@@ -175,6 +193,28 @@ func IsMultiFSEnabled() bool {
 	return false
 }
 
+// SetPoolSize method sets the sizes for MetadataPool and dataPool
+func SetPoolSize(context *clusterd.Context, clusterName, fsName string, dataPool []*model.Pool, metadataPool *model.Pool) error {
+	// updating the MetaDataPool size
+	metadataPool = GenerateMetadataPoolName(fsName, metadataPool)
+	err := SetPoolProperty(context, clusterName, metadataPool.Name, "size", strconv.Itoa(int(metadataPool.ReplicatedConfig.Size)))
+	if err != nil {
+		return errors.Wrapf(err, "failed to update metadataPool replica for filesystem %q", fsName)
+	}
+	// updating dataPool size
+	dataPool = GenerateDataPoolNames(fsName, dataPool)
+	for _, pool := range dataPool {
+		if pool == nil {
+			continue
+		}
+		err = SetPoolProperty(context, clusterName, pool.Name, "size", strconv.Itoa(int(pool.ReplicatedConfig.Size)))
```

Comment:
Something doesn't seem right if we're adding new code for setting the pool size property. For example:
- Erasure coded pools wouldn't be handled
- There is already code for creating pools for a new filesystem. We should be able to trigger than code in the orchestration instead of writing new code. Do you see where the pools are configured for a new filesystem?

Reply:
> Something doesn't seem right if we're adding new code for setting the pool size property. For example:
> 
> * Erasure coded pools wouldn't be handled

So I have to take care of the Erasure coded pools too, noted.
> * There is already code for creating pools for a new filesystem. We should be able to trigger than code in the orchestration instead of writing new code. Do you see where the pools are configured for a new filesystem?

Instead of writing a new function, just modify the existing function where pools were getting created?.

Reply:
> Something doesn't seem right if we're adding new code for setting the pool size property. For example:
> 
> * Erasure coded pools wouldn't be handled
I didn't get this, how am I ignoring the EC pools, say I have 10 pools which are both EC and non-EC. If they are being passed through a function to change their pool size, what am I missing in it?.
> * There is already code for creating pools for a new filesystem. We should be able to trigger than code in the orchestration instead of writing new code. Do you see where the pools are configured for a new filesystem?


JHel

Reply:
Hey Travis, I didn't get that, how am I missing the EC pools exactly, If there's a list of pools, how am I missing any of them regardless of they are EC or not.

cc @travisn  @leseb

Reply:
@rajatsingh25aug I took a closer look at this. I'm thinking we can follow the same pattern as in the pool CR, which calls `client.CreatePoolWithProfile()` every time it reconciles. If the settings change, the pool will be updated. This will update the replication size and other properties like the targetSizeRatio, and also handle the EC pools. So the filesystem code should be able to iterate through the pools and call that method, rather than setting the pool size.